subprojects/build-operations/src/main/java/org/gradle/trace/buildops/BuildOperationTrace.java

```diff
@@ -37,11 +47,11 @@ public BuildOperationTrace measureConfigurationTime(File outFile) {
     }
 
     public BuildOperationTrace measureBuildOperation(String buildOperationName, File outFile) {
-        Provider<BuildOperationDurationRecordingListener> listenerProvider = sharedServices.registerIfAbsent("build-op-" + buildOperationName, BuildOperationDurationRecordingListener.class, spec -> {
-            spec.getParameters().getCapturedBuildOperation().set(buildOperationName);
-            spec.getParameters().getOutputFile().set(outFile);
+        Provider<BuildOperationDurationRecordingListener> listenerProvider = sharedServices.registerIfAbsent("measure-build-operations", BuildOperationDurationRecordingListener.class, spec -> {
+            spec.getParameters().getCapturedBuildOperations().set(capturedBuildOperations);
         });
         registry.onOperationCompletion(listenerProvider);
+        capturedBuildOperations.put(buildOperationName, outFile);
```

Comment:
I was expecting to find this `put` before the registration.

Reply:
In the design you propose it would make sense that way, yes.